src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -148,45 +148,96 @@ private void DoWorkExpectCancel(object sender, DoWorkEventArgs e)
             manualResetEvent3.Set();
         }
 
+        #endregion
+
         [Fact]
         public void TestThrowExceptionInDoWork()
         {
-            var bw = new TestWorker();
-            bw.DoWork += DoWorkWithException;
-            bw.RunWorkerAsync("Exception");
+            const string expectedArgument = "Exception";
+            const string expectedExceptionMsg = "Exception from DoWork";
+            var bw = new BackgroundWorker();
+
+            bw.DoWork += (sender, e) =>
+            {
+                Assert.Same(bw, sender);
+                Assert.Same(expectedArgument, e.Argument);
+                throw new TestException(expectedExceptionMsg);
+            };
+
+            bw.RunWorkerCompleted += (sender, e) =>
+            {
+                TestException ex = Assert.Throws<TestException>(() => e.Result);
+                Assert.Equal(expectedExceptionMsg, ex.Message);
+            };
+
+            bw.RunWorkerAsync(expectedArgument);
```

Comment:
Shouldn't we block or otherwise prevent the fact method from completing until the test has completed?